server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2366,6 +2375,7 @@ public long getPersistedLocalCheckpoint() {
      */
     protected final void markSeqNoAsSeen(long seqNo) {
         localCheckpointTracker.advanceMaxSeqNo(seqNo);
+        lastOpPrimaryTerm = engineConfig.getPrimaryTermSupplier().getAsLong();
```

Comment:
I think we should advance the `maxPrimaryTerm` using the term of the operation instead of the current term of the replication group.